File: iosApp/iosAppTests/UIComponent/Nutrition/NutritionStateRobot.swift
import ComposeApp

final class NutritionStateRobot: BaseStateRobot<NutritionUiState, NutritionEvent> {

    override func defaultState() -> NutritionUiState {
        NutritionUiState(
            url: "https://example.test/nutrition",
            eventSink: createEventSink()
        )
    }

    func loadingState() -> NutritionUiState {
        NutritionUiState(
            url: nil,
            eventSink: createEventSink()
        )
    }
}

File: features/nutrition/impl/presentation/src/androidDeviceTest/kotlin/com/jkjamies/sampleplatter/features/nutrition/presentation/NutritionStateRobot.kt
package com.jkjamies.sampleplatter.features.nutrition.presentation

import com.jkjamies.sampleplatter.core.test.StateRobot

class NutritionStateRobot : StateRobot<NutritionUiState, NutritionEvent>() {

    override fun defaultState() = NutritionUiState(
        url = "https://example.test/nutrition",
        eventSink = createEventSink(),
    )

    fun loadingState() = NutritionUiState(
        url = null,
        eventSink = createEventSink(),
    )
}
